compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -765,7 +767,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         captured_projs: Vec<&[Projection<'tcx>]>,
     ) -> bool {
         let needs_drop = |ty: Ty<'tcx>| {
-            ty.needs_drop(self.tcx, self.tcx.param_env(closure_def_id.expect_local()))
+            ty.has_significant_drop(self.tcx, self.tcx.param_env(closure_def_id.expect_local()))
```

Comment:
Update the doc string on this function for
a) Remove the FIXME: and turn the associated info with now we only handle significant drops ...
b) In the call stack call NeedsSignificantDrop instead of NeedsDrop